test/core/stream/column_t.cpp

```diff
@@ -0,0 +1,162 @@
+
+#define BOOST_TEST_DYN_LINK
+#define BOOST_TEST_MODULE stream_column_t_test
+
+#include "common.hpp"
+
+#include <hipipe/core/stream/column_t.hpp>
+
+#include <boost/test/unit_test.hpp>
+
+// TODO add some tests!
+// Don't forget:
+// - Read doc/mainpage.md and doc/installation.md.
+// - See doc/installation.md for instructions how to run one or all unit tests.
+// - Test all the public functions.
+// - Fix the docstrings if they are wrong.
+// - common.hpp contains multiple predefined columns.
+// - Look into other tests if unsure how to proceed, e.g. for_each.cpp.
+// - Test whether move-only (e.g., std::unique_ptr) values work as well.
+// - Creating move-only data can sometimes be tricky,
+//   see e.g. for_each.cpp:test_for_each_move_only.
+// - Add yourself to the list of authors.
+// - Remove this comment.
+// - Make PR targeting into branch `runtime`.
+
+BOOST_AUTO_TEST_CASE(test_extract_column)
+{
+    using hipipe::stream::abstract_column;
+
+    std::unique_ptr<Int> col = std::make_unique<Int>();
+    col->data().assign({1, 2, 3});
+    BOOST_TEST(col->extract<Int>() == std::vector<int>({1, 2, 3}));
+
+    std::unique_ptr<abstract_column> ab_col = std::move(col);
+    BOOST_TEST(ab_col->extract<Int>() == std::vector<int>({1, 2, 3}));
+}
+
+
+BOOST_AUTO_TEST_CASE(test_extract_move_only_column)
+{
+    using hipipe::stream::abstract_column;
+
+    std::unique_ptr<Unique> col = std::make_unique<Unique>();
+    col->data().push_back(std::make_unique<int>(1));
+    col->data().push_back(std::make_unique<int>(2));
+    col->data().push_back(std::make_unique<int>(3));
+    BOOST_TEST(*(col->extract<Unique>().at(0)) == 1);
+    BOOST_TEST(*(col->extract<Unique>().at(1)) == 2);
+    BOOST_TEST(*(col->extract<Unique>().at(2)) == 3);
+
+    std::unique_ptr<abstract_column> ab_col = std::move(col);
+    BOOST_TEST(*(ab_col->extract<Unique>().at(0)) == 1);
+    BOOST_TEST(*(ab_col->extract<Unique>().at(1)) == 2);
+    BOOST_TEST(*(ab_col->extract<Unique>().at(2)) == 3);
+}
+
+
+BOOST_AUTO_TEST_CASE(test_extract_throws_error)
+{
+    using hipipe::stream::abstract_column;
+
+    std::unique_ptr<Int> col = std::make_unique<Int>();
+    col->data().assign({1, 2, 3});
+    BOOST_CHECK_THROW(col->extract<Double>(), std::runtime_error);
+
+    std::unique_ptr<abstract_column> ab_col = std::move(col);
+    BOOST_CHECK_THROW(ab_col->extract<Double>(), std::runtime_error);
+}
+
+
+BOOST_AUTO_TEST_CASE(test_get_column_size)
+{
+    Int col1;
+    col1.data().assign({1, 2, 3});
+    Unique col2;
+    BOOST_TEST(col1.size() == 3);
+    BOOST_TEST(col2.size() == 0);
+}
+
+
+BOOST_AUTO_TEST_CASE(test_take_column)
+{
+    using hipipe::stream::abstract_column;
+    
+    std::unique_ptr<Int> col1 = std::make_unique<Int>();
+    col1->data().assign({1, 2, 3, 4, 5});
+    std::unique_ptr<abstract_column> col2 = col1->take(2);
+    BOOST_TEST(col1->extract<Int>() == std::vector<int>({3, 4, 5}));
+    BOOST_TEST(col2->extract<Int>() == std::vector<int>({1, 2}));
```

Comment:
I like to test corner cases, can you also test taking all the elements, please? :-)